STUDIO/org.openl.rules.webstudio/src/org/openl/rules/rest/AclCommandSupport.java

```diff
@@ -0,0 +1,205 @@
+            return AclPermission.ALL_SUPPORTED_DEPLOY_CONFIG_REPO_PERMISSIONS;
+        } else if (RepoType.PROD == repoType) {
+            return AclPermission.ALL_SUPPORTED_PROD_REPO_PERMISSIONS;
+        } else {
+            throw new IllegalArgumentException("Unknown repo type: " + repoType);
+        }
+    }
+
+    public static AclCommand toCommand(String line, String deployConfigRepo) throws CommandFormatException {
+        String[] split = line.split(":");
+        if (line.endsWith(":")) {
+            String[] t = new String[split.length + 1];
+            System.arraycopy(split, 0, t, 0, split.length);
+            t[t.length - 1] = "";
+            split = t;
+        }
+        if (split.length <= 1) {
+            throw new CommandFormatException(MSG1);
+        }
+        Action action = toAction(split[0].trim());
+        if (split.length != 3 && (Action.LIST == action || Action.LIST_ALL == action) || split.length != 6 && (Action.ADD == action || Action.REMOVE == action || Action.SET == action)) {
+            throw new CommandFormatException(MSG1);
+        }
+        RepoType repoType = toRepoType(split[1].trim());
+        String repo = toRepo(split[1].trim());
+        if (StringUtils.isBlank(repo) && !StringUtils.isBlank(deployConfigRepo) && RepoType.DEPLOY_CONFIG == repoType) {
+            repo = deployConfigRepo;
+        }
+        String resource = split[2].trim();
+        if (repo == null && !resource.isEmpty()) {
+            throw new CommandFormatException(MSG4);
+        }
+        if (Action.ADD == action || Action.REMOVE == action || Action.SET == action) {
+            SidType sidType = toSidType(split[3].trim());
+            String sid = split[4].trim();
+            String[] permissions = split[5].trim().split(",");
+            permissions = Arrays.stream(permissions)
+                .map(String::trim)
+                .map(String::toUpperCase)
+                .filter(e -> !e.isEmpty())
+                .toArray(String[]::new);
+            if (permissions.length == 0 && Action.SET != action) {
+                throw new CommandFormatException(MSG2);
+            }
+            for (String permission : permissions) {
+                if (listAllSupportedPermissions(repoType).stream()
+                    .noneMatch(e -> Objects.equals(AclPermission.toString(e), permission.toUpperCase()))) {
+                    throw new CommandFormatException(String.format(MSG3, permission, repoType.getName()));
+                }
+            }
+            return new AclCommand(action, sidType, sid, repoType, repo, resource, permissions);
+        } else {
+            return new AclCommand(action, null, null, repoType, repo, resource, null);
+        }
+    }
+
+    private static String toRepo(String repo) {
+        int index = repo.indexOf("/");
+        if (index > 0) {
+            repo = repo.substring(index + 1).trim();
+        } else {
+            return null;
```

Comment:
Is it OK for `"/repo"` or just `"/"`?

Reply:
Yep, the pattern "design/design_id" is used. 0 is skipped.